generic-sensor/generic-sensor-tests.js

```diff
@@ -32,6 +32,40 @@ function reading_to_array(sensor) {
   return arr;
 }
 
+function load_iframe(src) {
+  return new Promise(resolve => {
+    const iframe = document.createElement("iframe");
+    iframe.onload = () => { resolve(iframe); };
+    iframe.srcdoc = src;
+    iframe.style.display = "none";
+    document.documentElement.appendChild(iframe);
+  });
+}
+
+function wait_for_message(iframe) {
+  return new Promise(resolve => {
+    self.addEventListener("message", function listener(e) {
```

Comment:
This optimization refers to [`idbfactory-open-opaque-origin.html`](https://github.com/w3c/web-platform-tests/blob/master/IndexedDB/idbfactory-open-opaque-origin.html), I think the idea is to clean up the listener once it receives the message to avoid secondary listening.